### ISSUE TITLE
I have tested it just now, and it Not works correctly with the latest 3.2 preview (which is currently being built as 3.2.0-rc2 for deployment to NuGet).

### ISSUE THREAD

**Issue by zesmaili** [elsa-workflows/elsa-core#5786]
I have tested it just now, and it works correctly with the latest 3.2 preview (which is currently being built as 3.2.0-rc2 for deployment to NuGet).

I tested with this custom activity:

```csharp
public class MyVariableSetterActivity : CodeActivity
{
    protected override void Execute(ActivityExecutionContext context)
    {
        context.SetVariable("Foo", "baz");
    }
}
```

In my workflow, I defined a variable called `Foo` with a default value of `X`.
The workflow looks like this:

<img width="2550" alt="image" src="https://github.com/elsa-workflows/elsa-core/assets/938393/6b108824-7aa8-48b2-94a6-53a5bdd02a5a">

Since I am unable to reproduce the reported symptom in the latest version, I will close this bug. However, please feel free to reopen if you can provide a reproduction using the latest (preview) version.

_Originally posted by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/issues/5653#issuecomment-2211292667_

**Comment by zesmaili** [comment]
> ```
>           I have tested it just now, and it works correctly with the latest 3.2 preview (which is currently being built as 3.2.0-rc2 for deployment to NuGet).
> ```
> 
> I tested with this custom activity:
> 
> ```cs
> public class MyVariableSetterActivity : CodeActivity
> {
>     protected override void Execute(ActivityExecutionContext context)
>     {
>         context.SetVariable("Foo", "baz");
>     }
> }
> ```
> 
> In my workflow, I defined a variable called `Foo` with a default value of `X`. The workflow looks like this:
> 
> <img alt="image" width="2000" src="https://private-user-images.githubusercontent.com/938393/346206667-6b108824-7aa8-48b2-94a6-53a5bdd02a5a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyMTU2NTIsIm5iZiI6MTcyMTIxNTM1MiwicGF0aCI6Ii85MzgzOTMvMzQ2MjA2NjY3LTZiMTA4ODI0LTdhYTgtNDhiMi05NGE2LTUzYTViZGQwMmE1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxN1QxMTIyMzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOWU2ZjY1NzI5OTc3NWY3ZTJlNDBmNzEzYWVjZWI1MjY3MmVmZTZmZTRmM2I5NGUyZDBjNTYxYjY3ZTdlM2ZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Hlf-mqMid6JXOTaRsptzd0ealUIy3dQ6Ar-oRmi2rQo">
> Since I am unable to reproduce the reported symptom in the latest version, I will close this bug. However, please feel free to reopen if you can provide a reproduction using the latest (preview) version.
> 
> _Originally posted by @sfmskywalker in [#5653 (comment)](https://github.com/elsa-workflows/elsa-core/issues/5653#issuecomment-2211292667)_

 @sfmskywalker You have not used the RunTask activity The problem is when we resume the workflow, in this case the value of the variables is not accessible

**Comment by sfmskywalker** [comment]
Correct, when workflows get resumed, variables are lost, unless they are Workflow Definition variables configured with the Workflow Storage Provider. You can configure this from the Variables section in the designer when editing or creating a workflow variable. When a variable is configured with a storage provider, the variable values will remain intact.

**Comment by zesmaili** [comment]
> Correct, when workflows get resumed, variables are lost, unless they are Workflow Definition variables configured with the Workflow Storage Provider. You can configure this from the Variables section in the designer when editing or creating a workflow variable. When a variable is configured with a storage provider, the variable values will remain intact.

@sfmskywalker
Can this be changed to also store in the workflow instance?
In Elsa 2, variables were stored in the workflow instance and it was much easier to access values ​​stored in variables.
Also, it's easier and simpler if we want to create a history of how and when the value of a variable changes.

**Comment by sfmskywalker** [comment]
Yes, if you use the Workflow Storage Driver, it will be stored in the workflow instance.
I agree that in V2 it’s much easier to fetch a workflow variable value, and we should probably implement an API to make it easy for consumer code in V3.
